ale/transformation.py

```diff
@@ -279,21 +278,25 @@ def generate_rotations(self, frames, times, time_bias, rotation_type, sensor_fra
         stop_et = max(times) 
         quats_and_avs_per_frame = []
 
-        logger.debug(f"Sensor times: {times}")
-        
-        for s, d in frames:
-            if exact_ck_times and self.use_web: 
-                function_args = {"startEt": start_et+time_bias, "stopEt": stop_et+time_bias, "toFrame": d, "refFrame": s, "mission": mission, "searchKernels": self.search_kernels}
-                quats_and_avs_per_frame.append(spiceql_call("getExactTargetOrientations", function_args, self.use_web))
-            else: 
-                with ThreadPoolExecutor(max_workers=3) as executor:
-                    futures = []
-                    for s, d in frames:
-                        function_args = {"toFrame": d, "refFrame": s, "mission": mission, "searchKernels": self.search_kernels}
-                        futures.append(executor.submit(get_ephem_data, times, "getTargetOrientations", 150, self.use_web, function_args))
-                    quats_and_avs_per_frame = [future.result() for future in futures]
+        logger.debug(f"Generate rotation times: {times}")
+        with ThreadPoolExecutor(max_workers=3) as executor:
+            futures = []
+
+            for s, d in frames:
+                if exact_ck_times and len(times) > 1:
+                    function_args = {"startEt": start_et+time_bias, "stopEt": stop_et+time_bias, "toFrame": d, "refFrame": s, "mission": mission, "searchKernels": self.search_kernels, "ckQualities": ["smithed", "reconstructed"], "fullKernelPath": False}
+                    futures.append(executor.submit(spiceql_call, "getExactTargetOrientations", function_args, self.use_web))
```

Comment:
this code cant really run on none-web spice. Some kernel sets will need more than one CK, causing this to fail since kernels are furnished in the __enter__ call. Considering that GetTartgetOrientations only causes a speed up in the web condition, I think it's fine limiting to web.